lib/internal/crypto/scrypt.js

```diff
@@ -107,8 +109,13 @@ function check(password, salt, keylen, options) {
       p = options.parallelization;
     }
     if (options.maxmem !== undefined) {
-      validateUint32(options.maxmem, 'maxmem');
       maxmem = options.maxmem;
+      if (typeof maxmem !== 'number')
+        throw new ERR_INVALID_ARG_TYPE('maxmem', 'number', maxmem);
+      if (!Number.isInteger(maxmem))
+        throw new ERR_OUT_OF_RANGE('maxmem', 'an integer', maxmem);
+      if (maxmem < 0 || maxmem > Number.MAX_SAFE_INTEGER)
```

Comment:
Should `Number` come from `primordials`? I suppose you could also open-code it as `maxmem >= 2 ** 53` since that's what you're doing on the line below.